webapp/channels/src/components/channel_header/channel_header_text_popover.tsx

```diff
@@ -0,0 +1,191 @@
+        handleClose: safePolygon({
+            requireIntent: false,
+            blockPointerEvents: true,
+        }),
+    });
+    const focus = useFocus(floatingContext);
+    const dismiss = useDismiss(floatingContext);
+    const click = useClick(floatingContext);
+    const role = useRole(floatingContext, {role: 'tooltip'});
+
+    const {getReferenceProps, getFloatingProps} = useInteractions([hover, focus, click, dismiss, role]);
+
+    const rootRef = useMergeRefs([rootElementRef, setReference]);
+
+    const maxWidthOfPopover = getMaxWidthOfPopover(rootElementRef?.current);
+
+    // This action processes clicks on formatted text elements like hashtags, user mentions,
+    // channel mentions, etc. while also allowing other elements to function as is such as external links etc
+    function handleClick(event: MouseEvent<HTMLDivElement>) {
+        dispatch(handleFormattedTextClick(event, currentRelativeTeamUrl));
+    }
+
+    return (
+        <>
+            <div
+                ref={rootRef}
+                className='header-description__text'
+                {...getReferenceProps()}
+            >
+                <Markdown
+                    message={props.text}
+                    options={markdownOptions.inHeader}
+                    imageProps={IMAGE_MARKDOWN_OPTIONS}
+                />
+            </div>
+
+            {isMounted && (
+                <FloatingPortal id={RootHtmlPortalId}>
+                    <FloatingOverlay
+                        className='channel-header-text-popover-floating-overlay'
+                        lockScroll={true}
+                    >
+                        {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
+                        <div
+                            ref={setFloating}
+                            className='channel-header-text-popover'
+                            style={{
+                                maxWidth: maxWidthOfPopover,
+                                ...floatingStyles,
+                                ...transitionStyles,
+                            }}
+                            onClick={handleClick}
+                            {...getFloatingProps()}
+                        >
+                            <Markdown
+                                message={props.text}
+                                options={markdownOptions.inPopover}
+                                imageProps={IMAGE_MARKDOWN_OPTIONS}
+                            />
+                        </div>
+                    </FloatingOverlay>
+                </FloatingPortal>
+            )}
+        </>
+    );
+}
+
+function checkIfTextIsOverflowing(elem: HTMLDivElement | null, text: string): boolean {
+    if (!elem) {
+        return false;
+    }
+
+    if (text.match(/\n{2,}/g)) {
+        return true;
+    }
+
+    if (elem.scrollWidth === elem.clientWidth && elem.scrollHeight === elem.clientHeight) {
```

Comment:
I think that checking these width/height values while React is rendering can cause the JS thread to block waiting for the UI to be repainted. I'm not certain if that's actually a concern, but it's something to keep an eye on

Reply:
yes we can check, but since the ref and element for a channel would remain same i am not concerned but let me try to move this to a useEffect